core/core-configuration-layer.el

```diff
@@ -295,41 +299,61 @@ cache folder.")
   (configuration-layer/load-or-install-protected-package 'package-build)
   (configuration-layer/load-or-install-protected-package 'quelpa))
 
+(defun configuration-layer//resolve-package-archive (archive)
+  "Resolve HTTP handlers for ARCHIVE.
+If the address of an ARCHIVE already contains the protocol then this address is
+left untouched."
+  (cons (car archive)
+        (if (or (string-match-p "http" (cdr archive))
+                (string-prefix-p "/" (cdr archive)))
+            (cdr archive)
+          (concat (if (and dotspacemacs-elpa-https
+                           (not spacemacs-insecure)
+                           ;; for now org ELPA repository does
+                           ;; not support HTTPS
+                           ;; TODO when org ELPA repo support
+                           ;; HTTPS remove the check
+                           ;; `(not (equal "org" (car archive)))'
+                           (not (equal "org" (car archive))))
+                      "https://"
+                    "http://")
+                  (cdr archive)))))
+
 (defun configuration-layer//resolve-package-archives (archives)
   "Resolve HTTP handlers for each archive in ARCHIVES and return a list
 of all reachable ones.
 If the address of an archive already contains the protocol then this address is
 left untouched.
 The returned list has a `package-archives' compliant format."
-  (mapcar
-   (lambda (x)
-     (cons (car x)
-           (if (or (string-match-p "http" (cdr x))
-                   (string-prefix-p "/" (cdr x)))
-               (cdr x)
-             (concat
-              (if (and dotspacemacs-elpa-https
-                       (not spacemacs-insecure)
-                       ;; for now org ELPA repository does
-                       ;; not support HTTPS
-                       ;; TODO when org ELPA repo support
-                       ;; HTTPS remove the check
-                       ;; `(not (equal "org" (car x)))'
-                       (not (equal "org" (car x))))
-                  "https://"
-                "http://")
-              (cdr x)))))
-   archives))
+  (mapcar 'configuration-layer//resolve-package-archive archives))
+
+(defun configuration-layer//check-archive-status (archive)
+  "Check ARCHIVE status.
+
+Available return values are available and unavailable."
+  (let ((url (format "%sarchive-contents"
+                     (cdr (assoc archive package-archives))))
+        (state 'unavailable))
+    (condition-case nil
+        (when (or (and (string-prefix-p "/" url)
+                       (file-readable-p url))
```

Comment:
@d12frosted Where can we find the documentation about `file-readable-p` capable of checking URLs except in the C source ? This function does not block for a long time when the URL is not reachable ? What kind of tests did you do to check that `file-readable-p` is not blocking for a long time ?